### DISCUSSION TITLE
How to visually select entire line upwards with x & v?

### DISCUSSION THREAD

**Discussion by txtyash** [helix-editor/helix#3775] [Q&A] [answered]
```
0 Brown is a fruit
1 Orange is a color |
2 Fox is an animal
```
In the above code block, `|`(bar) represents the cursor and the numbers represent line numbers. Now if I have to select the lines 1 & 2 then I can press `xx` but if I have to select lines 1 & 0 upwards then `xv` is pretty much useless as it does not select the line 1. I can select those lines with `vghk`. Why is `xx` functionality only available downwards? It should at least be working with `xvk`. Gravity?

*Basically is there no simple `V` like key as in vim*

I come from simple `v`, `V` & `ctrl+v` in vim so I'm unable to understand some stuff like:
```
% | Select entire file | select_all
x | Select current line, if already selected, extend to next line | extend_line_below
X | Extend selection to line bounds (line-wise selection) | extend_to_line_bounds
Alt-x | Shrink selection to line bounds (line-wise selection) | shrink_to_line_bounds
```
And how things work with `x` & `v` here. Can someone please explain that? Especially, `V` & `ctrl+v` from vim

**Comment by the-mikedavis** [comment, marked as answer]
You can use `vkX`. `vk` will create a selection that includes something on lines 0 and 1 and then `X` expands the selection to be linewise.

There is also a `extend_line_above` command which doesn't have a default binding that behaves like `x` (`extend_line_below`) but goes upwards. Plus there's a newer command `extend_line` which works like `x` but extends in the direction of the selection. If you move upwards a lot then you may want to bind one of these.

**Reply by txtyash** [reply]
Thanks but those bindings don't seem intuitive like other helix bindings

**Comment by MaxBeaud** [comment]
I've found this to be intuitive. 

```toml
[keys.normal]
X = "select_line_above"
```
This binds `X` to select the whole line upwards which nicely complement `x` which selects the whole line downwards

**Reply by ardijanr** [reply]
Nice one, tried this and it also works with ``X10X`` which selects 10 lines upwards. Great stuff!
